src/main/scala/amailp/intellij/robot/parser/package.scala

```diff
@@ -18,6 +18,12 @@ package object parser {
     def currentType = getTokenType
     def currentText = getTokenText
 
+    def parserCommentLine(includingTerminator: Boolean = true): Unit = {
+        while (currentType != SettingsHeader) {
```

Comment:
I believe this check is fruitless, since is already done in `RobotParser` inside the `parseTableItemsWithSubParser` call